src/AutofacSerilogIntegration/ContextualLoggingModule.cs

```diff
@@ -48,8 +50,17 @@ protected override void Load(ContainerBuilder builder)
 
                 return logger;
             })
-            .As<ILogger>()
-            .ExternallyOwned();
+            .As<ILogger>();
+
+            if (_dispose)
+            {
+                registration.SingleInstance()
```

Comment:
I don't think this will have the desired effect - no individual loggers's lifetime is going to correspond with the overall container lifetime, and each instance of this component may be a type-specific `ForContext()` logger, so not necessarily usable for subsequent activations.

I think we'll need a separate (named?) registration for:

```csharp
var logger = _logger ?? Log.Logger;
```

I.e., it'd become:

```csharp
var logger = c.ResolveNamed<ILogger>("AutofacSerilogIntegration.Logger");
```

and it's this named registration that would get the release logic.